### PR TITLE
New package: gpu-burn

### PR DESCRIPTION
**Pull request by glennpj**
The gpu-burn package installs the gpu_burn utility for stress testing
Nvidia GPUs.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gpu-burn/package.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    url      = "http://wili.cc/blog/entries/gpu-burn/gpu_burn-1.0.tar.gz"
+
+    version('1.0', sha256='d55994f0bee8dabf021966dbe574ef52be1e43386faeee91318dd4ebb36aa74a')
+
+    patch('Makefile.patch')
+
+    # This package uses CudaPackage to pick up the cuda_arch variant. A side
+    # effect is that it also picks up the cuda variant, but cuda is required
+    # for gpu-burn so is not really a variant.
+    variant('cuda', 'True')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpu-burn/package.py]: This is an interesting use case that we didn't really think about when we wrote the `CudaPackage`.

Can you copy the description to the variant?

File: var/spack/repos/builtin/packages/gpu-burn/package.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    conflicts('~cuda', msg='gpu-burn requires cuda')
+
+    def patch(self):
+        # update cuda architecture if necessary
+        if '+cuda' in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpu-burn/package.py]: No need for if given the conflict above.

File: var/spack/repos/builtin/packages/gpu-burn/package.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    )
+                archflag = '-arch=compute_{0}'.format(cuda_arch[0])
+
+            filter_file('-arch=compute_30', archflag,
+                        'Makefile', string=True)
+
+    def install(self, spec, prefix):
+        mkdir(prefix.bin)
+        mkdir(prefix.share)
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpu-burn/package.py]: I would switch this to a `MakefilePackage` base class, then this line won't be needed anymore. `patch` can be renamed to `edit` as well.

**Comment by glennpj** [comment]
@adamjstewart Is there anything else that needs to be done for this PR?

**Comment by scheibelp** [comment]
The comment at https://github.com/spack/spack/pull/13108#discussion_r333091905 would simplify this but IMO is not required for merging

**Pull request merged**